lib/manager/poetry/extract.ts

```diff
@@ -84,3 +85,23 @@ function extractFromSection(
   });
   return deps;
 }
+
+function extractRegistries(pyprojectfile: PoetryFile): string[] {
+  const sources =
+    pyprojectfile.tool &&
+    pyprojectfile.tool.poetry &&
+    pyprojectfile.tool.poetry.source;
+
+  if (!Array.isArray(sources) || sources.length === 0) {
+    return null;
+  }
+
+  const registryUrls: string[] = ['https://pypi.org/pypi/'];
+  for (const source of sources) {
+    if (source.url) {
+      registryUrls.push(source.url);
```

Comment:
Should we de-dupe this in case the user configures pypi.org explicitly?

Reply:
Sorry, I misread this comment as a statement instead of a question. I committed code that dedupes, but I don't think it is necessary as pypi datasource will only append pypi.org if registryUrls is empty. If poetry manager adds pypi.org itself, pypi datasource won't add any other datasource. Let me revert this.